Source/Core/VideoCommon/OnScreenUI.h

```diff
@@ -61,6 +61,9 @@ class OnScreenUI
 
 private:
   void DrawDebugText();
+#ifdef USE_RETRO_ACHIEVEMENTS
+  void DrawChallenges();
```

Comment:
Mentioned this in the chat but this shouldn't be owned by the `OnScreenUI`, it should be owned by `AchievementManager`.

Should do something like:

```
const auto achievement_manager = AchievementManager::GetInstance();
achievement_manager->Display();
```

instead and have achievement manager know how to display the challenge icons.  This keeps the logic out of `OnScreenUI` which should stay somewhat generic.

Reply:
Yes, it should, but that makes the texture destruction logic a lot more complicated. I'd recommend we come together and refactor this at some point in the future.

Reply:
Yeah, you can see how I do this for the editor:  https://github.com/dolphin-emu/dolphin/blob/16aa85447e4488564d91aedc45b023fa0ad118cf/Source/Core/VideoCommon/OnScreenUI.cpp#L115 .  I assumed you wanted to punt on this but wanted to call this out in a more public place.  Thanks!

Reply:
Made a first-pass attempt at migrating this into AchievementManager. Now if only I could get it to work on all builders...